index.js

```diff
@@ -536,20 +539,12 @@ RedisClient.prototype.return_error = function (err) {
         this.should_buffer = false;
     }
 
-    if (command_obj && typeof command_obj.callback === "function") {
-        try {
-            command_obj.callback(err);
-        } catch (callback_err) {
-            // if a callback throws an exception, re-throw it on a new stack so the parser can keep going
-            process.nextTick(function () {
-                throw callback_err;
-            });
-        }
-    } else {
-        console.log("node_redis: no callback to send error: " + err.message);
-        // this will probably not make it anywhere useful, but we might as well throw
+    try {
+        command_obj.callback(err);
+    } catch (callback_err) {
+        // if a callback throws an exception, re-throw it on a new stack so the parser can keep going
         process.nextTick(function () {
-            throw err;
+            throw callback_err;
```

Comment:
It is safe to remove the case that the callback is not a function (I could not reproduce that behavior) as it should never happen. But even if it would happen it is safe to remove the special handling as it will also be catched by the try catch and rethrown async.